.pre-commit.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env sh
+# pre-commit.sh
+
+STASH_NAME="pre-commit-$(date +%s)"
+RED='\033[1;31m'
+GREEN='\033[1;32m'
+YELLOW='\033[1;33m'
+NC='\033[0m'
+
+if git stash save -u -k -q $STASH_NAME; then
+    echo "${YELLOW}Stashed changes as:${NC} ${STASH_NAME}\n\n"
+    stash=1
```

Comment:
@torkleyy I figured it out, if there are no local changes to stash, `git stash save` exits without error and sets `stash` to 1; later the script attempts to unstash something I saved from another branch causing unmerged paths.  

That doesn't explain why it doesn't happen if my commit isn't empty though.